crypto/async/async_wait.c

```diff
@@ -148,6 +148,33 @@ int ASYNC_WAIT_CTX_clear_fd(ASYNC_WAIT_CTX *ctx, const void *key)
             continue;
         }
         if (curr->key == key) {
+            /* If fd has just been added, remove it from the list */
+            if (curr->add == 1) {
+                if (ctx->fds == curr) {
+                    ctx->fds = curr->next;
+                }
+                else {
+                    struct fd_lookup_st *prev = ctx->fds;
+                    while (prev->next != curr && prev->next != NULL) {
```

Comment:
Yes, this is definitely possible. I'll add the changes to remove the second loop